tests/Credentials/GCECredentialsTest.php

```diff
@@ -124,6 +126,51 @@ public function testOnGceWithResidency()
         $this->assertTrue(GCECredentials::onGCE($httpHandler));
     }
 
+    public function testOnWindowsGceWithResidencyWithNoCom()
+    {
+        if (class_exists(COM::class)) {
+            throw $this->markTestSkipped('This test in meant to handle when the COM extension is not preset');
+        }
+
+        $method = (new ReflectionClass(GCECredentials::class))
+            ->getMethod('detectResidencyWindows');
+        
+        $method->setAccessible(true);
+
+        $this->assertFalse($method->invoke(null, 'thisShouldBeFalse'));
+    }
+
+    public function testOnWindowsGceWithResidencyNotOnGCE()
+    {
+        if (!class_exists(COM::class)) {
+            throw $this->markTestSkipped('This test only works while running on windows COM extension enabled');
+        }
+
+        $keyPathProperty = 'HKEY_LOCAL_MACHINE\\SYSTEM\\HardwareConfig\\Current\\';
+        $keyName = 'SystemProductName';
+
+        $method = (new ReflectionClass(GCECredentials::class))
+            ->getMethod('detectResidencyWindows');
+        $method->setAccessible(true);
+
+        $this->assertFalse($method->invoke(null, $keyPathProperty . $keyName));
+    }
+
+    public function testOnWindowsGceWithResidency()
+    {
+        throw $this
+            ->markTestSkipped('This test only works while running on windows COM class enabled and running on GCE');
```

Comment:
I agree with the Windows one but the last bit seems a bit odd. We are using the code that we are testing to determine if we should test it.

If there is something wrong with the code, the code would avoid the test to run and viceversa 🤔 What do you think @bshaffer

Reply:
Yeah I agree it's a bit weird.

The idea is that we only run it manually, so we'd know if it was supposed to be skipped or not. If it skips when we're running it on windows GCE, then we know it's broken.

The way it's coded now, it will fail on GA actions - the above way it would at least skip as expected, and we could run it on our own.

The other option I see is to assert it's a bool - so it runs and passes, but we don't actually verify that it's true. This is more useful for CI but less useful for testing the actual feature (which we still want to do manually)

Reply:
Hmmm yeaaaah, if we just test is a bool is kind of stepping on the previous test, the NotOnGCE one, which is testing is false but it wouldn't be a real test.

All right, let's do it with the weird pattern, here is my train of thought:
If we still need to run this feature test manually, there is no real benefit to assert is boolean if in the end, we still need to run this manually. I prefer having this weird pattern and being able to run it manually vs adding another test and also keeping a manual one, I do not see the benefit of that.